apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TableSchemaBuilder.cs

```diff
@@ -40,10 +41,30 @@ public sealed class TableSchemaBuilder : IEnumerable
         public void Add(TableFieldSchema field)
         {
             GaxPreconditions.CheckNotNull(field, nameof(field));
-
+            _tableFields[field.Name] = field;
             _fields.Add(field);
         }
 
+        /// <summary>
+        /// Modifies the given fields as per the specified action.
+        /// </summary>
+        /// <param name="fieldName">The name of the field to be modified. Must not be null.</param>
+        /// <param name="modifyField">An Action to be performed on the given field.</param>
+        /// <returns>A <see cref="TableSchemaBuilder"/> with the modified field.</returns>
+        public TableSchemaBuilder ModifyField(string fieldName, Action<TableFieldSchema> modifyField)
+        {
+            GaxPreconditions.CheckNotNull(fieldName, nameof(fieldName));
```

Comment:
We should validate that `modifyField` (or whatever it's going to be called) isn't null, either.

Reply:
done